tests/conftest_phase31.py

```
@@ -0,0 +1,443 @@
+                errors.append(f"Field {field} cannot be None")
+
+        # Type validations
+        if "id" in doc and not isinstance(doc["id"], int):
+            errors.append("Field 'id' must be integer")
+
+        if "keywords" in doc and not isinstance(doc["keywords"], list):
+            errors.append("Field 'keywords' must be list")
+
+        if "confidence_score" in doc:
+            score = doc["confidence_score"]
+            if not isinstance(score, int | float) or not (0.0 <= score <= 1.0):
```

Pylint (3.3.7)

[convention] Unnecessary parens after 'not' keyword

(C0325)